src/strategy/exponential_backoff.rs

```diff
@@ -1,14 +1,16 @@
 use std::time::Duration;
 use std::iter::Iterator;
-use std::u64::{MAX as U64_MAX};
+use std::u64::MAX as U64_MAX;
 
 /// A retry strategy driven by exponential back-off.
 ///
 /// The power corresponds to the number of past attempts.
-#[derive(Clone)]
+#[derive(Debug, Clone)]
 pub struct ExponentialBackoff {
     current: u64,
-    base: u64
+    base: u64,
+    factor: Option<u64>,
```

Comment:
I actually had this at first, but then I thought if user the default behavior (factor of 1 / no factor) then you would wind up incurring the `checked_mul` cost for each `next` iteration even if the factor itself is not doing anything. Switching it to an `Option` avoids this cost. Probably minimal difference either way.

I can definitely change it though if you think it's better that way.